test/components/structures/TimelinePanel-test.tsx

```diff
@@ -513,4 +516,64 @@ describe("TimelinePanel", () => {
             replyToEvent.mockClear();
         });
     });
+
+    it("renders when the last message is an undecryptable thread root", () => {
+        jest.spyOn(SettingsStore, "getValue").mockImplementation((name) => name === "feature_threadstable");
+
+        const client = MatrixClientPeg.get();
+        client.isRoomEncrypted = () => true;
+        client.supportsExperimentalThreads = () => true;
+        client.decryptEventIfNeeded = () => Promise.resolve();
+        const authorId = client.getUserId()!;
+        const room = new Room("roomId", client, authorId, {
+            lazyLoadMembers: false,
+        });
+
+        const events = mockEvents(room);
+        const props = {
+            ...getProps(room, events),
+        };
+
+        const { rootEvent } = mkThread({
+            room,
+            client,
+            authorId,
+            participantUserIds: [authorId],
+        });
+
+        events.push(rootEvent);
+
+        const roomMembership = mkMembership({
+            mship: "join",
+            prevMship: "join",
+            user: authorId,
+            room: room.roomId,
+            event: true,
+        });
+        roomMembership.event.state_key = "123";
+
+        events.push(roomMembership);
```

Comment:
is including this rather weird-looking membership event in the timeline important for the test? why?

Also, it means that "the last message is an undecryptable thread root" is not the case: the thread root is the penultimate message

Reply:
Good shout! I've moved the event to be at the start of the timeline rather than being the last message.